src/excursion/Excursion.svelte

```diff
@@ -0,0 +1,105 @@
+<script>
+  import { excursion } from "../store";
+  import { Collection } from "sveltefire";
+  const addToArray = (provisionRef) => {
+    provisionRef.add({});
+  };
+
+  const deleteFromArray = (provisionRef, id) => {
+    provisionRef
+      .doc(id)
+      .delete()
+      .then((r) => console.log(r));
+  };
+</script>
+
+<div>
+  <Collection
+    path={$excursion.collection('provisions')}
+    let:data={excursionsData}
+    let:ref>
+    <form on:submit|preventDefault={() => addToArray(ref)}>
+      <ul class="divide-y divide-gray-200">
+        {#each excursionsData as item}
+          <li class="px-4 py-4 sm:px-6">
+            <div>
+              <label
+                for="price"
+                class="block text-sm font-medium text-gray-700">Price</label>
+              <div class="mt-1 relative rounded-md shadow-sm">
+                <div
+                  class="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
+                  <span class="text-gray-500 sm:text-sm"> $ </span>
+                </div>
+                <input
+                  value={item.price || ''}
+                  on:change={({ target: { value } }) => {
+                    ref.doc(item.id).update({ price: value });
+                  }}
+                  type="text"
+                  name="price"
+                  id={`price-${item.id}`}
+                  class="focus:ring-indigo-500 focus:border-indigo-500 block w-full pl-7 pr-12 sm:text-sm border-gray-300 rounded-md"
+                  placeholder="0.00" />
+                <div class="absolute inset-y-0 right-0 flex items-center">
+                  <label for="currency" class="sr-only">Currency</label>
+                  <select
+                    value={item.currency}
+                    on:blur={({ target: { value } }) => {
+                      ref.doc(item.id).update({ currency: value });
+                    }}
+                    id={`currency-${item.id}`}
+                    name="currency"
+                    class="focus:ring-indigo-500 focus:border-indigo-500 h-full py-0 pl-2 pr-7 border-transparent bg-transparent text-gray-500 sm:text-sm rounded-md">
+                    <option>USD</option>
```

Comment:
should be another file eventually, but fine for now. since we're going to be using this probably multiple places

Reply:
I agreee. My though process was to extract it as soon as we needed it in a second place.